### PR TITLE
Refactor UI & Storage to accomodate Project and Issue

### LINKED ISSUE #50: Refactor UI to support showing projects, clients and issues

### PR DESCRIPTION
**Pull request by ConradLew**
Fixes #50

### CODE REVIEW

File: src/main/java/seedu/address/model/project/Deadline.java [hunk 1]
```diff
@@ -32,6 +32,8 @@ private EmptyDeadline() {
 
     private LocalDate projectDeadline;
 
+    private String deadlineString;
```
Comment (Yongbeom-Kim) [inline src/main/java/seedu/address/model/project/Deadline.java:35]: Not sure why additional deadline string is stored when it can be fetched from the projectDeadline object - but not a major thing

**Review by Yongbeom-Kim — COMMENTED** [review]
Other than regex, lgtm

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 2]
```diff
@@ -41,6 +42,7 @@ public Command parseCommand(String userInput) throws ParseException {
         }
 
         final String commandWord = matcher.group("commandWord");
+        final String flag = matcher.group("flag"); // returns a whitespace, followed by the tag e.g. " -f".
         final String arguments = matcher.group("arguments");
```
Comment (Yongbeom-Kim) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:46]: In addition, I would like to request you to please parse arguments into a proper format other than a string (HashMap, maybe?) - or else this logic of parsing arguments will have to be repeated in all CommandParser classes. If you would prefer it done in a separate PR/by me, lmk as well.

**Review by Yongbeom-Kim — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java (hunk 2 above)
Comment (Yongbeom-Kim) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:46]: In addition, I would like to request for you to please parse arguments into a proper format before passing it onto the CommandParser classes, or else this logic will have to be repeated in all the commandparser classes. LMK if you would prefer this done in a separate PR or by someone else

**Review by Yongbeom-Kim — CHANGES_REQUESTED** [review]

**Review by Yongbeom-Kim — CHANGES_REQUESTED** [review]
I should request changes for this too huh
EDIT: :( mistake i can't delete this so yall just have to deal with it

File: src/main/java/seedu/address/model/AddressBook.java [hunk 3]
```diff
@@ -85,6 +143,30 @@ public void setPerson(Person target, Person editedPerson) {
         persons.setPerson(target, editedPerson);
     }
 
+    /**
+     * Replaces the given person {@code target} in the list with {@code editedProject}.
```
Comment (crvstalphua) [inline src/main/java/seedu/address/model/AddressBook.java:147]: should be given project instead of given person?
```suggestion
     * Replaces the given project {@code target} in the list with {@code editedProject}.
```

File: src/main/java/seedu/address/model/AddressBook.java [hunk 4]
```diff
@@ -85,6 +143,30 @@ public void setPerson(Person target, Person editedPerson) {
...
+     * existing project in the address book.
+     */
+    public void setProject(Project target, Project editedProject) {
+        requireNonNull(editedProject);
+
+        projects.setProject(target, editedProject);
+    }
+
+    /**
+     * Replaces the given person {@code target} in the list with {@code editedIssue}.
```
Comment (crvstalphua) [inline src/main/java/seedu/address/model/AddressBook.java:159]: should be given issue instead of given person?
```suggestion
     * Replaces the given issue {@code target} in the list with {@code editedIssue}.
```

**Review by crvstalphua — APPROVED** [review]
Not sure if javadocs has been done yet but found two minor typos for previously done javadocs. Other than that, LGTM!

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 5]
```diff
@@ -25,7 +25,8 @@ public class AddressBookParser {
     /**
      * Used for initial separation of command word and args.
      */
-    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+    private static final Pattern BASIC_COMMAND_FORMAT =
+            Pattern.compile("(?<commandWord>\\S+)(?<flag>(\\s-\\S+)?)(?<arguments>.*)");
```
[the remark this answers is not in the document]
Comment (ConradLew) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java]: Thanks for the suggestions! I've changed `\S` to `\w` and `\s` to `\s+` as recommended. 

I'm unable to move the `\s`  out of the named flag capture group as it seems to remove the whitespace from the beginning of the arguments group in non-flag commands, which causes a bug in the parser for the arguments group (For some reason, the provided parser requires that the argument group starts with a whitespace). 

I have provided a temporary fix by trimming off the whitespace at the start of the flag group; we can look to fixing the given arguments group parser at a later date.

With regards to commands without arguments, I've tested and the current regex seems to work fine for that case.

File: src/main/java/seedu/address/model/project/Deadline.java (hunk 1 above)
Comment (ConradLew) [reply src/main/java/seedu/address/model/project/Deadline.java]: Change made

File: src/main/java/seedu/address/model/AddressBook.java (hunk 3 above)
Comment (ConradLew) [reply src/main/java/seedu/address/model/AddressBook.java]: Change made

**Review by Donovan9617 — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java (hunk 2 above)
Comment (ConradLew) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java:46]: Could I clarify what you mean by a proper format?

Comment (Yongbeom-Kim) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java:46]: Sorry, please ignore - this should be done in another PR

**Review by Yongbeom-Kim — APPROVED** [review]
lgtm!
